SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -366,3 +378,304 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
+  VelInfMod = sqrt(VelInfMod);
+
+  /*--- Compute the recovery factor ---*/
+  // Double-check: laminar or turbulent Pr for this?
+  su2double Recovery = pow(config->GetPrandtl_Lam(), (1.0/3.0));
+
+  /*--- Typical constants from boundary layer theory ---*/
+
+  const su2double kappa = 0.41;
+  const su2double B = 5.5;
+  const su2double cv1_3 = 7.1*7.1*7.1;
+
+  bool converged = true;
+
+  for (auto iMarker = 0u; iMarker < config->GetnMarker_All(); iMarker++) {
+
+    if ((config->GetMarker_All_KindBC(iMarker) == HEAT_FLUX) ||
+        (config->GetMarker_All_KindBC(iMarker) == ISOTHERMAL) ) {
+
+      /*--- Identify the boundary by string name ---*/
+
+      string Marker_Tag = config->GetMarker_All_TagBound(iMarker);
+
+      /*--- Jump to another BC if it is not wall function ---*/
+
+      if (config->GetWallFunction_Treatment(Marker_Tag) != STANDARD_WALL_FUNCTION)
+        continue;
+
+      /*--- Get the specified wall heat flux from config ---*/
+
+      su2double q_w = config->GetWall_HeatFlux(Marker_Tag);
+
+      /*--- Loop over all of the vertices on this boundary marker ---*/
+
+      for (auto iVertex = 0u; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+        const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+        const auto Point_Normal = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
+
+        /*--- Check if the node belongs to the domain (i.e, not a halo node)
+         and the neighbor is not part of the physical boundary ---*/
+
+
+          /*--- Get coordinates of the current vertex and nearest normal point ---*/
+
+          su2double *Coord = geometry->nodes->GetCoord(iPoint);
+          su2double *Coord_Normal = geometry->nodes->GetCoord(Point_Normal);
+
+          /*--- Compute dual-grid area and boundary normal ---*/
+
+          // getnormal and not normal_neighbor
+          const auto Normal = geometry->vertex[iMarker][iVertex]->GetNormal();
+
+          geometry->vertex[iMarker][iVertex]->GetNormal(Vector);
+          su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+
+          su2double UnitNormal[MAXNDIM] = {0.0};
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            UnitNormal[iDim] = -Normal[iDim]/Area;
+
+          /*--- Get the velocity, pressure, and temperature at the nearest
+           (normal) interior point. ---*/
+          su2double Vel[MAXNDIM] = {0.0};
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            Vel[iDim] = nodes->GetVelocity(Point_Normal,iDim);
+
+          P_Normal = nodes->GetPressure(Point_Normal);
+          T_Normal = nodes->GetTemperature(Point_Normal);
+
+          /*--- Compute the wall-parallel velocity at first point off the wall ---*/
+
+          //su2double VelNormal = 0.0;
+          //for (auto iDim = 0u; iDim < nDim; iDim++)
+          //  VelNormal += Vel[iDim] * UnitNormal[iDim];
+     
+          su2double VelNormal = GeometryToolbox::DotProduct(int(MAXNDIM), Vel, UnitNormal);
```

Comment:
why `nDim` in the Norm above and here `int MAXNDIM`. Sure it makes the size known at compile time (i.e. maybe a bit faster) or are there other reasons?

Reply:
I did that in some other places too, it was a bit of a micro optimization.

Reply:
Yes, I copied a line of code from elsewhere, is there really a difference in the end?

Reply:
I personally wouldn't worry about this ... but maybe Pedro is preparing some big openFOAM-like 3D-everything push to get rid of all those run-time evaluated nDim loops 🤔 :D

Reply:
For sure
```
template<size_t nDim>
class SU2 {

};
```